### ISSUE TITLE
Getting started with a custom dataset and "next suggested experiment"

### ISSUE THREAD

**Issue by sgbaird** [janosh/thermo#15]
Nice work! Could you give some pointers or a README tutorial on how to use `thermo` with a custom dataset for materials discovery, and suggest the next experiment?

Which `.py` file would you recommend starting out with?

I'm assuming the code is general to material inputs and properties rather than specifically thermoelectrics, correct?

**Comment by janosh** [comment]
Thanks for your interest!

> Could you give some pointers or a README tutorial on how to use `thermo` with a custom dataset for materials discovery, and suggest the next experiment?

I should probably archive this repo at some point as it might give the false impression that this is a library of tools ready for use on other problems. It's just a collection of experiments really.

> Which `.py` file would you recommend starting out with?

The random forest is fastest to train and conceptually simplest. Multiplicative Normalizing Flow might be worth taking a look at too since it surprisingly beat the RF both i.t.o. accuracy and quality of uncertainty estimates even on our very small datasets where RF usually excels. I compared these two models in [`mnf_vs_rf.py`](https://github.com/janosh/thermo/blob/main/notebooks/leaderboard/mnf_vs_rf/mnf_vs_rf.py).

> I'm assuming the code is general to material inputs and properties rather than specifically thermoelectrics, correct?

Yes.

**Comment by sgbaird** [comment]
Thanks for addressing this. I may have been under a misconception about the repo 😅. Based on another issue, you mentioned that you used a hand-curated set of Magpie descriptors. In another issue, you also mentioned some of the UQ methods. Did you have an acquisition function (e.g. expected improvement) that you maximized to get a next suggested experiment? With your clarifications so far, I'm probably also at a point where I can dig into some of the files (e.g. normalizing flow) a bit more. I think a big part was knowing where to look.

EDIT: in the linked `.py` file from above, I'm seeing that the Adam optimizer is used, at least for normalizing flow. does `n_pred` refer to the number of adaptive design iterations or something else?

**Comment by janosh** [comment]
`n_pred` is the number of predictions we're asking the NF to make. Making predictions with a flow involves sampling from the base distribution and "flowing" that sample through the chain of bijections (that were tuned during training) into a sample for the target distribution, every prediction will be slightly different. We can then compute statistics across all `n_pred` predictions to estimate the overall predictive uncertainty. Since sampling multiple times from the base distribution is a vectorized operation where all samples are drawn in parallel, this is very fast.
